### PR TITLE
Use setMap / getMap in Layer objects

### PR DESCRIPTION
**Pull request by jorgelamb**
Hi,

I send a pull request with changes in several Layer objects. It fixes the removal of the layers from the map.

There was a similar commit (https://github.com/branflake2267/GWT-Maps-V3-Api/commit/eccd442740ce657a055ffa1dde92d2f7c8d7bd4a) for PanoramioLayer, this includes all the other layers.

At first I thought Options objects needed the same change but after running the tests I found they didn't.

Thanks for the library, it's very useful! Keep up the good work.

Hope my first pull request is ok! :)

Jorge

### CODE REVIEW

File: Apis_Google_Maps/src/com/google/gwt/maps/client/adsense/AdUnitOptions.java [hunk 1]
```diff
@@ -100,11 +100,15 @@ private final native void setFormatImpl(String format) /*-{
    * @param mapWidget
    */
   public final void setMap(MapWidget mapWidget) {
-    setMapImpl(mapWidget.getJso());
+    if (mapWidget == null) {
```
Comment (branflake2267) [inline Apis_Google_Maps/src/com/google/gwt/maps/client/adsense/AdUnitOptions.java:103]: If this is going to be used like this, setting widget to be null, the widget has to be removed from parent to I think.

Comment (branflake2267) [inline Apis_Google_Maps/src/com/google/gwt/maps/client/adsense/AdUnitOptions.java:103]: I'm trying to remember if I did this too

**Comment by branflake2267** [comment]
Do the changes work for you?

File: Apis_Google_Maps/src/com/google/gwt/maps/client/adsense/AdUnitOptions.java (hunk 1 above)
Comment (branflake2267) [inline Apis_Google_Maps/src/com/google/gwt/maps/client/adsense/AdUnitOptions.java:103]: Nevermind me, its early in the morning. Looks good.

**Comment by branflake2267** [comment]
Looks good to me. I'm going to merge. :) Thanks. :)

File: Apis_Google_Maps/src/com/google/gwt/maps/client/adsense/AdUnitOptions.java (hunk 1 above)
Comment (jorgelamb) [inline Apis_Google_Maps/src/com/google/gwt/maps/client/adsense/AdUnitOptions.java:103]: :) They work for me, yes. At least for weather, transit, bicycle, ...

**Pull request merged**